### ISSUE TITLE
[BUG] Optimize FastAI hyperparameter throws exception: `unsupported operand type(s) for *: 'NoneType' and 'float'`

### ISSUE THREAD

**Issue by se-jaeger** [awslabs/autogluon#2202]
**Describe the bug**
I'm using the 0.5.3b20221010 pre-release because it fixed a bug I ran into. Unfortunately, it introduces another one.

Optimizing the default hyperparameters (local random search without setting `time_limit` but `num_trials=50`) of a FastAI model results in the following exception stack trace:
````
Warning: Exception caused NeuralNetFastAI to fail during hyperparameter tuning... Skipping this model.
Traceback (most recent call last):
  File "/usr/local/lib/python3.9/site-packages/autogluon/core/trainer/abstract_trainer.py", line 1600, in _train_single_full
    hpo_models, hpo_results = model.hyperparameter_tune(
  File "/usr/local/lib/python3.9/site-packages/autogluon/core/models/abstract/abstract_model.py", line 1000, in hyperparameter_tune
    return self._hyperparameter_tune(hpo_executor=hpo_executor, **kwargs)
  File "/usr/local/lib/python3.9/site-packages/autogluon/core/models/abstract/abstract_model.py", line 1033, in _hyperparameter_tune
    trial_soft_time_limit = max(hpo_executor.time_limit * 0.9, hpo_executor.time_limit - 5)  # 5 seconds max for buffer
TypeError: unsupported operand type(s) for *: 'NoneType' and 'float'
unsupported operand type(s) for *: 'NoneType' and 'float'
````

**My dive into the codebase**
Even after initializing the `hpo_executor` object, its `time_limit` attribute is still `None` causing the above exception. Since this line was introduced after the 0.5.2 version, I assume this is OK but it's necessary to catch this edge case.
Compare https://github.com/awslabs/autogluon/blob/master/core/src/autogluon/core/models/abstract/abstract_model.py#L1044 and https://github.com/awslabs/autogluon/blob/0.5.2/core/src/autogluon/core/models/abstract/abstract_model.py#L1042

I'm not sure why `trial_soft_time_limit` was added so I can't file a PR fixing this but I'm happy to push this forward if someone gives me more background on this.

**Comment by gradientsky** [comment]
Reproducible example:
```python
from autogluon.tabular import TabularDataset, TabularPredictor
if __name__ == '__main__':
    train_data = TabularDataset('https://autogluon.s3.amazonaws.com/datasets/Inc/train.csv')
    subsample_size = 500  # subsample subset of data for faster demo, try setting this to much larger values
    train_data = train_data.sample(n=subsample_size, random_state=0)
    train_data.head()
    label = 'class'
    print("Summary of class variable: \n", train_data[label].describe())
    save_path = 'agModels-predictClass'  # specifies folder to store trained models
    hyperparameters = {'FASTAI': {}}
    hyperparameter_tune_kwargs = {  # HPO is not performed unless hyperparameter_tune_kwargs is specified
        'num_trials': 10,
        'scheduler': 'local',
        'searcher': 'auto',
    }
    predictor = TabularPredictor(label=label, path=save_path).fit(train_data, hyperparameters=hyperparameters, hyperparameter_tune_kwargs=hyperparameter_tune_kwargs)
    test_data = TabularDataset('https://autogluon.s3.amazonaws.com/datasets/Inc/test.csv')
    y_test = test_data[label]  # values to predict
    test_data_nolab = test_data.drop(columns=[label])  # delete label column to prove we're not cheating
    test_data_nolab.head()
    y_pred = predictor.predict(test_data_nolab)
    print("Predictions:  \n", y_pred)
    perf = predictor.evaluate_predictions(y_true=y_test, y_pred=y_pred, auxiliary_metrics=True)
```

**Comment by Innixma** [comment]
This is fixed in #2263 to my knowledge, planned to be merged soon
